crypto/src/main/kotlin/org/kethereum/crypto/Signatures.kt

```diff
@@ -1,10 +1,19 @@
 package org.kethereum.crypto
 
+import org.kethereum.extensions.hexToBigInteger
 import org.kethereum.extensions.toHexStringNoPrefix
 import org.kethereum.extensions.toHexStringZeroPadded
 import org.kethereum.model.SignatureData
+import org.komputing.khex.model.HexString
 import java.math.BigInteger
 
+fun HexString.toSignatureData(): SignatureData {
```

Comment:
I think we should rather make this a constructor to SignatureData than an extension function to HexString

Reply:
My proposition is to do it with factory method then:
```kotlin
companion object {
    fun fromHex(hexSignature: String): SignatureData {
        val cleanedHex = HexString(hexSignature).clean0xPrefix().string
        val r = HexString(cleanedHex.substring(0, 64)).hexToBigInteger()
        val s = HexString(cleanedHex.substring(64, 128)).hexToBigInteger()
        val v = HexString(cleanedHex.substring(128)).hexToBigInteger()
        return SignatureData(r, s, v)
    }
}
```
Then call it like `SignatureData.fromHex("0x...")`

Why I don't like the constructor is because as I see it we'll need to embrace mutability:
```kotlin
constructor(hexSignature: String) : this() {
    val cleanedHex = HexString(hexSignature).clean0xPrefix().string
    r = HexString(cleanedHex.substring(0, 64)).hexToBigInteger()
    s = HexString(cleanedHex.substring(64, 128)).hexToBigInteger()
    v = HexString(cleanedHex.substring(128)).hexToBigInteger()
}
```
We're initializing `SignatureData` with 0s and then setting `rsv`. Looks like a code smell to me.

Reply:
makes sense - thanks! Yea just did not like the extension to HexString as extending very broad objects like this leads to unusable autocompletion at some point - but this is great now - thanks for the contribution!